tests/test_recv_timeout.py

```diff
@@ -49,7 +49,7 @@ def test_recv_timeout_vs_keepalive(self) -> None:
                 socket_mock.recv_into.assert_called()
 
                 now = time.monotonic()
-                assert recv_timeout <= (now - start) < keep_alive
+                assert recv_timeout <= round(now - start, 2) <= keep_alive
```

Comment:
Instead of rounding, I'd suggest adding a small fudge factor to `keep_alive`, something like:
```suggestion
                assert recv_timeout <= (now - start) <= (keep_alive + 0.1)
```
@FoamyGuy does that make sense to you?